lib/child_process.js

```diff
@@ -236,7 +237,7 @@ function exec(command, options, callback) {
 }
 
 const customPromiseExecFunction = (orig) => {
-  return (...args) => {
+  return assignFunctionName(orig.name, function(...args) {
```

Comment:
Wouldn't we want to also re-assign `length` property? It of course doesn't need to be in this PR, but throwing it out there in case we want to add that later

Reply:
Ideally yes, but I'm not quite sure if sacrificing DRY and inlining `customPromiseExecFunction()` with everything properly named wouldn't be a better solution in this particular place on a long term.
If there are another places that would require setting `length` manually, later we can adjust `assignFunctionName()` and add this feature, since it's internal-only.